hibernate-core/src/main/java/org/hibernate/collection/spi/AbstractPersistentCollection.java

```diff
@@ -282,7 +282,11 @@ else if ( !session.isConnected() ) {
 				}
 			}
 			else {
-				if ( !session.isTransactionInProgress() ) {
+				/*
+				 Whenever the collection lazy loading is triggered during the loading process,
+				 closing the connection will cause an error when RowProcessingStateStandardImpl#next() will be called.
+				 */
+				if ( !session.isTransactionInProgress() && session.getPersistenceContext().isLoadFinished() ) {
```

Comment:
Wow, that is so much better and more comprehensible if that really works. Fingers crossed that tests run through 🤞

Reply:
Thanks for all your suggestions they led to this solution , I run it locally and for H2 all tests passed so :crossed_fingers:

Reply:
@beikov can I merge it?

Reply:
@beikov Thanks